test/sys/test_sockopt.rs

```diff
@@ -1,9 +1,9 @@
 use rand::{thread_rng, Rng};
-use nix::sys::socket::{socket, sockopt, getsockopt, setsockopt, AddressFamily, SockType, SockFlag, SockLevel};
+use nix::sys::socket::{socket, sockopt, getsockopt, setsockopt, AddressFamily, SockType, SockFlag, SockProtocol};
 
 #[test]
 fn test_so_buf() {
-    let fd = socket(AddressFamily::Inet, SockType::Datagram, SockFlag::empty(), SockLevel::Udp as i32).unwrap();
+    let fd = socket(AddressFamily::Inet, SockType::Datagram, SockFlag::empty(), SockProtocol::Udp).unwrap();
```

Comment:
Maybe we should use `SockProtocol::Default` (0) here as well as it seems that `SOCK_DGRAM` only expect the protocol to be zero ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))